p256/fiat_p256.mli

```diff
@@ -25,7 +26,8 @@ type scalar
 
 val scalar_of_cs : Cstruct.t -> (scalar, string) result
 (** Read data from a cstruct.
-    It should be 32 bytes long, in big endian format. *)
```

Comment:
The convention for error cases in Mirage libraries is to use the Rresult convention of 
```
`Error (`Msg "foo")
```

rather than a direct string. That'll make it easier to compose with other parts of the stack.

Reply:
nitpick: ``Error (`Msg "foo")``, not `` `Error (`Msg "foo")``, and yes, I agree in respect to easier composition.